code/clue.js

```diff
@@ -1,59 +1,363 @@
+  professorPlum,
+  missScarlet,
+  mrsPeacock,
+  colonelMustard,
   mrsWhite
-  // ...  and the rest
-]
+];
 
-const weapons = []
+const weapons = [
+  rope, 
+  knife, 
+  candlestick, 
+  dumbbell, 
+  poison, 
+  axe, 
+  bat, 
+  trophy, 
+  pistol
+];
 
-const rooms = []
+const rooms = [
+  diningRoom, 
+  conservatory, 
+  kitchen, 
+  study, 
+  library, 
+  billiardRoom, 
+  lounge, 
+  ballroom, 
+  hall, 
+  spa, 
+  livingRoom, 
+  observatory, 
+  theater,
+  guestHouse,
+  patio
+];
 
-// THIS FUNCTION WILL RANDOMLY SELECT ONE ITEM FROM THE ARRAY THAT YOU PASS IN TO THE FUNCTION.
-// YOU DON'T NEED TO CHANGE THIS, JUST TRY TO UNDERSTAND IT. AND HOW TO USE IT.
 const randomSelector = array => {
   return array[Math.floor(Math.random() * array.length)]
 }
 
-// CREATE AN OBJECT THAT KEEPS THE MYSTERY.
-// With a killer, a weapon and a room.
-// The values will be set later.
+// Mystery object
+let mystery = {
+  killer: '',
+  weapon: '',
+  room: ''
+};
 
-// This function will be invoked when you click on the killer card.
+//This shuffles the favourite weapon
+const shuffleFavouriteWeapon = () => {
+  suspects.forEach((suspect) => {
+  const randomWeapon = randomSelector(weapons);
+  suspect.favouriteWeapon = randomWeapon.id;
+  });
+};
+
+
+// Pick killer function
 const pickKiller = () => {
-  // This will randomly select a killer from the suspects. And add that to the mystery object.
-  mystery.killer = randomSelector(suspects)
-
-  // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
-  document.getElementById('killerCard').style.background = mystery.killer.color
-  document.getElementById(
-    'killerName'
-  ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
+    if (!killerIsPicked) {
+    document.getElementById('killerLoader').style.visibility='visible' 
+    setTimeout(function () {
+      document.getElementById('killerLoader').style.visibility='hidden'
+      shuffleFavouriteWeapon(suspects); //This randomly selects a favourite weapon.
+      mystery.killer = randomSelector(suspects);
+      document.getElementById('killerCard').style.background = mystery.killer.color;
+      document.getElementById('killerName').innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`;
+      document.getElementById('killerAge').innerHTML = mystery.killer.age;
+      document.getElementById('killerOccupation').innerHTML = mystery.killer.occupation;
+      document.getElementById('killerImage').src = mystery.killer.image;
+      document.getElementById('killerDescription').innerHTML = mystery.killer.description;
+      document.getElementById('favouriteWeapon').innerHTML = `Favourite weapon: ${mystery.killer.favouriteWeapon}`;
+      killerIsPicked = true;
+      }, 2500);
+    }
+};
+
+document.getElementById('killerCard').addEventListener('click', pickKiller);
```

Comment:
You don't need this line since you are using "onclick" in the HTML-file. Either you keep this in the JS-file and remove the "onclick" or vice versa.

Please update you PR and I will later on approve it!